app/models/warehouse_report/cas_declines.rb

```diff
@@ -0,0 +1,76 @@
+###
+# Copyright 2016 - 2019 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+class WarehouseReport::CasDeclines < OpenStruct
+  include ArelHelper
+
+  attr_accessor :start_date
+  attr_accessor :end_date
+
+  def initialize(start_date:, end_date:)
+    @start_date = start_date
+    @end_date = end_date
+  end
+
+  def reasons
+    (declines + cancels).map do|row|
+      reason = row.decline_reason || row.administrative_cancel_reason
+      reason.squish.gsub(/Other.*/,'Other').strip
+    end.each_with_object(Hash.new(0)) do |reason, counts|
+      counts[reason] += 1
+    end.sort_by(&:last).reverse
+  end
+
+  def declines
+    @declines ||= report_source.declined.
+      started_between(start_date: start_date, end_date: end_date + 1.day)
```

Comment:
Why `+ 1.day`?

Reply:
I believe it was to include items started on that day... though it was just a copy/paste from the previous version of the report.